src/PE/PEInfo.cs

```diff
@@ -77,35 +77,54 @@ sealed class PEInfo {
 		/// <param name="rva">The RVA</param>
 		/// <returns></returns>
 		public ImageSectionHeader ToImageSectionHeader(RVA rva) {
+			uint alignment = imageNTHeaders.OptionalHeader.SectionAlignment;
 			foreach (var section in imageSectionHeaders) {
-				if (rva >= section.VirtualAddress && rva < section.VirtualAddress + Math.Max(section.VirtualSize, section.SizeOfRawData))
+				if (rva >= section.VirtualAddress && rva < section.VirtualAddress + DotNet.Utils.AlignUp(section.VirtualSize, alignment))
 					return section;
 			}
 			return null;
 		}
 
 		/// <summary>
-		/// Converts a <see cref="FileOffset"/> to an <see cref="RVA"/>
+		/// Converts a <see cref="FileOffset"/> to an <see cref="RVA"/>, returns 0 if out of range
 		/// </summary>
 		/// <param name="offset">The file offset to convert</param>
 		/// <returns>The RVA</returns>
 		public RVA ToRVA(FileOffset offset) {
+			if (imageSectionHeaders.Length == 0)
+				return (RVA)offset;
+			// In pe headers
```

Comment:
It seems like your comments is about the previous line, could you add the comments before the statement, eg.
```suggestion
			if (imageSectionHeaders.Length == 0) {
				// In pe headers
				return (RVA)offset;
			}
```

I also added `{ }` since it now has two lines instead of one.

or 

```C#

			if ((uint)rva >= imageNTHeaders.OptionalHeader.SizeOfImage)
				return 0;
			// Check if rva is larger than memory layout size
->
			// Check if rva is larger than memory layout size
			if ((uint)rva >= imageNTHeaders.OptionalHeader.SizeOfImage)
				return 0;
```

Same thing about the other comments below.

Reply:
Done.